packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -2615,21 +2629,37 @@ describe('TransactionController', () => {
         userFeeLevel,
       });
 
-      const transaction = controller.state.transactions[0];
+      const transaction = controller.state.transactions.find(
+        ({ id }) => id === transactionId,
+      );
 
       expect(transaction?.txParams?.gas).toBe(gas);
       expect(transaction?.txParams?.gasLimit).toBe(gasLimit);
       expect(transaction?.txParams?.gasPrice).toBe(gasPrice);
-      expect(transaction?.txParams?.maxPriorityFeePerGas).toBe(
-        maxPriorityFeePerGas,
-      );
-      expect(transaction?.txParams?.maxFeePerGas).toBe(maxFeePerGas);
       expect(transaction?.estimateUsed).toBe(estimateUsed);
       expect(transaction?.estimateSuggested).toBe(estimateSuggested);
       expect(transaction?.defaultGasEstimates).toBe(defaultGasEstimates);
       expect(transaction?.originalGasEstimate).toBe(originalGasEstimate);
       expect(transaction?.userEditedGasLimit).toBe(userEditedGasLimit);
       expect(transaction?.userFeeLevel).toBe(userFeeLevel);
+
+      // In order to test maxPriorityFeePerGas, maxFeePerGas updates
+      // we shoudn't set the gasPrice in the same call to not fail the validation
+      controller.updateTransactionGasFees(txIdWithoutGasPriceUpdate, {
```

Comment:
This makes total sense, but would it be cleaner to split this into two separate unit tests?

Maybe `updates provided legacy gas values` and `updates provided 1559 gas values`?

Reply:
Splitted both in https://github.com/MetaMask/core/pull/1978/commits/3f1068adcc11074c7b815536518e44e6a827ab44